docs/proposals/nginx-extensions.md

```diff
@@ -0,0 +1,964 @@
+| Session persistence                           | Provisional  | [GEP-1619](https://gateway-api.sigs.k8s.io/geps/gep-1619/)               |
+| Rate-limiting                                 | Open         | [Issue 326](https://github.com/kubernetes-sigs/gateway-api/issues/326)   |
+| Cross-Origin Resource Sharing (CORS)          | Open         | [Issue 1767](https://github.com/kubernetes-sigs/gateway-api/issues/1767) |
+| Gateway client certificate verification       | Provisional  | [GEP-91](https://gateway-api.sigs.k8s.io/geps/gep-91/)                   |
+| Authentication                                | Open         | [Issue 1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494) |
+
+## Grouping the Features
+
+To reduce the number of CRDs NGINX Gateway Fabric must maintain and users have to create, we grouped the high and medium-priority features into configuration categories.
+
+For each group, we will begin implementation by focusing on the high-priority features of each group. Then, once the high-priority features are complete, we can move on to the medium-priority features.
+
+The low-priority features are out of scope for the rest of the Enhancement Proposal but may be revisited once we make progress on the higher-priority features. In addition, most features with active GEPs are not included in the groups as we want to help move the GEPs forward toward standardization instead of creating our bespoke solutions.
+
+When grouping the features, we considered the following factors:
+
+1. Use cases. Are there features that can be grouped by the use case they satisfy? For example, authentication or observability.
+2. Gateway API roles. Which features are the domain of the Cluster Operator? Which features are Application Developers responsible for?
+3. The NGINX contexts. Which contexts are the NGINX directives available in? For example, http, server, location, upstream, etc.
+
+The following picture shows the nine groups we came up with:
+
+![groups](/docs/images/nginx-functionality-groups.png)
+
+## API
+
+The following section proposes an extension type (Policy, Filter, etc.), extension point (GatewayClass, HTTPRoute, etc.), and Gateway API role for all the groups. The goal of this Enhancement Proposal is not to design each extension but to propose a strategy for delivering critical NGINX features to our customers.
+
+### Gateway Settings
+
+_Extension type:_ ParametersRef
+
+_Resource type:_ CRD
+
+_Role(s):_ Cluster Operator
+
+_Extension point:_ GatewayClass or Gateway
+
+_NGINX Context(s):_ main, http, stream
+
+Features:
+
+- Error log level
+- Access log settings: format and disable
+- PROXY protocol
+- OTEL Tracing: global settings such as exporter configuration and global enable/disable.
+- External Routing: enable/disable routing to ExternalName Services and set DNS resolver addresses.
```

Comment:
the resolver only makes sense for NGINX Plus - resolving for NGINX OSS is limited - you can't define an upstream. shameless plug -- https://www.nginx.com/blog/dns-service-discovery-nginx-plus/ :)

although we can in theory support routing to external destinations in NGINX OSS - we will just need to generate IPs for NGINX OSS -- basically write re-resolver in go...

Reply:
Good to know, and nice blog!

Should we only support this in Plus, then?

Reply:
> Should we only support this in Plus, then?

yep